tests/integration/main-test.js

```diff
@@ -273,4 +277,38 @@ module('Integration | main', function(hooks) {
   test('it sets sync relationships which were empty initially', async function(assert) {
     await testInitiallyEmptyRelationships.call(this, assert, 'sync-user');
   });
+
+  function unwrapProxy(o) {
+    return isProxy(o) ? unwrapProxy(get(o, 'content')) : o;
+  }
+
+  function isProxy(o) {
+    return !!(o && (o instanceof ObjectProxy || o instanceof ArrayProxy));
+  }
+
+  async function testBelongsToPresenceValidation(assert, userType) {
+    let user =  this.store.createRecord(userType);
+    function userValidator({ key, newValue }) {
+      const value = unwrapProxy(newValue);
+      if (key === 'profile') {
+        return isEmpty(value) ? 'Cannot be blank' : true;
```

Comment:
I believe `newValue` will either be CHANGES or CONTENT.  So if `profile` existed on the underlying model, then `newValue` should technically be "something" b/c it exists on the model.

Are you expecting `newValue` to be undefined?

Reply:
In case of sync belongsTo() it is null.
In case of async belongsTo() it should be null wrapped in ObjectProxy. There is an unwrapProxy() above, so at this point, it is already unwrapped, so it is null.

Reply:
Right good point.  At this point though, it is the ObjectTreeNode proxy, not your proxy.  Let me know what you think about the below PR!  Basically we need to unwrap before sending through to the validator function.

https://github.com/validated-changeset/validated-changeset/pull/61